suite-native/module-accounts/src/screens/AccountsScreen.tsx

```diff
@@ -1,15 +1,93 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import { View } from 'react-native';
 
-import { Screen } from '@suite-native/navigation';
-import { Box, Text } from '@suite-native/atoms';
+import { useFocusEffect } from '@react-navigation/native';
+
+import {
+    AccountsStackParamList,
+    AccountsStackRoutes,
+    Screen,
+    StackProps,
+} from '@suite-native/navigation';
+import { Box, Button, Chip } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { CryptoIcon } from '@trezor/icons';
+import { NetworkSymbol } from '@suite-common/wallet-config';
 
 import { AccountsList } from '../components/AccountsList';
 
-export const AccountsScreen = () => (
-    <Screen>
-        <Box marginBottom="medium">
-            <Text color="gray600">All Accounts</Text>
-        </Box>
-        <AccountsList />
-    </Screen>
-);
+const assetsFilterStyle = prepareNativeStyle(utils => ({
+    flexDirection: 'row',
+    borderRadius: utils.borders.radii.round,
+    marginBottom: utils.spacings.medium,
+}));
+
+const assetsFilterItemStyle = prepareNativeStyle(utils => ({
+    marginRight: utils.spacings.small,
+}));
+
+const discoverAssetsButtonStyle = prepareNativeStyle(utils => ({
+    borderRadius: utils.borders.radii.round,
+}));
+
+export const AccountsScreen = ({
+    route,
+}: StackProps<AccountsStackParamList, AccountsStackRoutes.Accounts>) => {
+    const { applyStyle } = useNativeStyles();
+    const [selectedAssets, setSelectedAssets] = useState<NetworkSymbol[]>([]);
+
+    const { currencySymbol } = route.params;
+
+    useFocusEffect(
+        useCallback(() => {
+            if (currencySymbol) {
+                setSelectedAssets([currencySymbol]);
+            }
+
+            return () => {
+                setSelectedAssets([]);
+            };
+        }, [currencySymbol]),
+    );
+
+    const handleSelectAsset = (currencySymbol: NetworkSymbol) => {
+        setSelectedAssets(assets => {
```

Comment:
This will be better and easier to understand
```tsx
        setSelectedAssets(assets => {
             if (assets.includes(currencySymbol)) {
                 return A.removeFirst(assets, currencySymbol)
             }
             return [...assets, currencySymbol];
         });
```

Maybe you can even replace:
```tsx
return [...assets, currencySymbol];
// replace with 
return A.append(assets, currencySymbol);
```

Which is also kind of more declarative than spread operator, but I will leave it to you, it's just small detail.

Reply:
I used your proposal with `includes` and replaced it with the `filter` instead of @mobily/ts-belt lib. I personally prefer vanilla JS if there is no other reason than syntax...

Reply:
It's not about syntax at all but about declarative vs imperative. Declarative approach is much better because you are saying only what should happen. Using imperative way you need to say what should happen and how it should happen and that is from definition harder to read and more messy.

For example:
```
assets.filter(asset => asset !== currencySymbol)
// vs
A.removeFirst(assets, currencySymbol)
```

As you can see in first case you how to define exact removal should happen, you need choose filter operator and then exactly define how this filter should behave. If you are reading it again you need to stop and think what that condition mean, will it filter all occurrences or just first one? And this is getting much more visible for complicated examples.

If you take second example, you can read it like a book. You see `A.removeFirst` and you don't even need think about it.